language/resource-api/README.md

```diff
@@ -176,58 +177,109 @@ class Puppet::Provider::AptKey::AptKey
 end
 ```
 
-The `get` method reports the current state of the managed resources. It returns an enumerable of all existing resources. Each resource is a hash with attribute names as keys, and their respective values as values. It is an error to return values not matching the type specified in the resource type. If a requested resource is not listed in the result, it is considered to not exist on the system. If the `get` method raises an exception, the provider is marked as unavailable during the current run, and all resources of this type will fail in the current transaction. The exception message will be reported to the user.
+The `get` method reports the current state of the managed resources. It returns an enumerable of all existing resources. Each resource is a hash with attribute names as keys, and their respective values as values.
+
+* It is an error to return values not matching the type specified in the resource type.
+* If a requested resource is not listed in the result, it is considered to not exist on the system.
+* If the `get` method raises an exception, the provider is marked as unavailable during the current run, and all resources of this type will fail in the current transaction. The exception message will be reported to the user.
+* If the type has more than one namevar, each resource hash must have a `:title` key with a single formatted string representing all namevar values. The title must match one of the title patterns. The namevar values computed from that pattern must match their counterparts in the hash.
+  > Note: This value allows the runtime environment to present resources in a way familiar to the user.
+
+  > Example: Referring back to the `software` example above, resources returned from the `get` method require a `title:` value matching the `package:` and `manager:` values:
+  > ```
+  > {
+  >     title: 'php-yum',
+  >     package: 'php',
+  >     manager: 'yum',
+  >     ensure: 'present',
+  > }
+  > ```
 
 The `set` method updates resources to a new state. The `changes` parameter gets passed a hash of change requests, keyed by the resource's name. Each value is another hash with the optional `:is` and `:should` keys. At least one of the two has to be specified. The values will be of the same shape as those returned by `get`. After the `set`, all resources should be in the state defined by the `:should` values.
 
+> Example for a `changes` request:
+> ```
+> changes = {
+>   'name' => {
```

Comment:
I think this example would be clearer if the name value was something like 'foo' . 
```
changes = {
  'foo' => {
    is: {
      name: 'foo',
      ensure: 'present',
      value: 'original value',
    },
    should: {
      name: 'foo',
      ensure: 'present',
      value: 'new value',
    },
  },
}
```